cng/ecdsa.go

```diff
@@ -237,10 +237,21 @@ func SignECDSA(priv *PrivateKeyECDSA, hash []byte) (r, s BigInt, err error) {
 
 // VerifyECDSA verifies the signature in r, s of hash using the public key, pub.
 func VerifyECDSA(pub *PublicKeyECDSA, hash []byte, r, s BigInt) bool {
-	if len(r) != len(s) {
+	// r and s might be shorter than size
+	// if the original big number contained leading zeros.
+	if len(r) > pub.size || len(s) > pub.size {
 		return false
 	}
-	sig := append(r, s...)
+	sig := make([]byte, 0, pub.size*2)
+	prependZeros := func(l int) {
+		if zeros := pub.size - l; zeros > 0 {
```

Comment:
This is silly, but... I'm a little leery of `l` as an identifier (especially as an `int`, where `l` could be substituted for a `1` and still compile).

```suggestion
	prependZeros := func(nonZeroBytes int) {
		if zeros := pub.size - nonZeroBytes; zeros > 0 {
```